ui/packages/catalog-ui-search/src/main/webapp/component/search-form-editor/search-form-editor.view.js

```diff
@@ -118,10 +123,50 @@ module.exports = Marionette.LayoutView.extend({
         selectionInterface: new SelectionInterface(),
         onMapLoaded: olMap => {
           this.showQueryForm(collection, id)
+          if (this.model.get('filterTemplate')) {
+            const geoFilters = (
+              this.model.get('filterTemplate').filters || [
+                this.model.get('filterTemplate'),
+              ]
+            ).filter(filter => CQLUtils.isGeoFilter(filter.type))
+            const coords = this.wktToCoords(geoFilters)
+            Common.queueExecution(() => {
+              this.map.currentView.map.zoomToExtent(coords)
+            })
+          }
         },
       })
     )
   },
+  wktToCoords(wktList) {
+    const parsedGeoCoords = _.flatten(
+      wktList.map(wkt => {
+        if (wkt.value.startsWith('POINT') && parseFloat(wkt.distance) !== 0) {
+          const center = terraformer.parse(wkt.value).coordinates
+          return new TurfCircle(center, parseFloat(wkt.distance), 64, 'meters')
+            .geometry.coordinates[0]
+        }
+        return terraformer.parse(wkt.value).coordinates
+      }),
+      true
+    )
+    return this.createCoordinatePairs(this.flatten(parsedGeoCoords))
+  },
+  flatten(arr) {
```

Comment:
❓ I'm curios why we're doing this when we're using the underscore `flatten` function here: https://github.com/codice/ddf/pull/6483/files#diff-61f3b5db651cfd7e289c041c39274cfa06a5b7e476b54fc20ef9ad5c574a21feR142 ?

Reply:
it lets you specify a second argument, `true`, meaning it will only flatten to one level. but it looks like `.flat()` can achieve the same thing: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat

so i will just use .flat()